tests/integration/src/xvm.rs

```diff
@@ -0,0 +1,334 @@
+        }
+    }
+}
+
+pub type CallResult = u32;
+
+#[ink::chain_extension]
+pub trait XvmCall {
+    type ErrorCode = u32;
+
+    #[ink(extension = 0x00010001, handle_status = false)]
+    fn call(vm_id: u8, target: Vec<u8>, input: Vec<u8>, value: u128) -> CallResult;
+}
+
+pub enum CustomEnvironment {}
+impl Environment for CustomEnvironment {
+    const MAX_EVENT_TOPICS: usize = <DefaultEnvironment as Environment>::MAX_EVENT_TOPICS;
+
+    type AccountId = <DefaultEnvironment as Environment>::AccountId;
+    type Balance = <DefaultEnvironment as Environment>::Balance;
+    type Hash = <DefaultEnvironment as Environment>::Hash;
+    type BlockNumber = <DefaultEnvironment as Environment>::BlockNumber;
+    type Timestamp = <DefaultEnvironment as Environment>::Timestamp;
+
+    type ChainExtension = XvmCall;
+}
+
+ */
+
+#[test]
+fn evm_payable_call_via_xvm_works() {
+    new_test_ext().execute_with(|| {
+        let evm_payable_addr = deploy_evm_contract(EVM_PAYABLE);
+
+        let value = UNIT;
+        assert_ok!(Xvm::call(
+            Context {
+                source_vm_id: VmId::Wasm,
+                weight_limit: Weight::from_parts(1_000_000_000, 1024 * 1024),
+            },
+            VmId::Evm,
+            ALICE,
+            evm_payable_addr.as_ref().to_vec(),
+            // Calling `deposit`
+            hex::decode("d0e30db0").expect("invalid selector hex"),
+            value,
+        ));
+        assert_eq!(
+            Balances::free_balance(account_id_from(evm_payable_addr)),
+            value
+        );
+
+        assert_ok!(Xvm::call(
+            Context {
+                source_vm_id: VmId::Wasm,
+                weight_limit: Weight::from_parts(10_000_000_000, 1024 * 1024),
+            },
+            VmId::Evm,
+            ALICE,
+            evm_payable_addr.as_ref().to_vec(),
+            // `Calling withdraw`
+            hex::decode("3ccfd60b").expect("invalid selector hex"),
+            0,
+        ));
+        assert_eq!(
+            Balances::free_balance(account_id_from(evm_payable_addr)),
+            ExistentialDeposit::get(),
+        );
+    });
+}
+
+#[test]
+fn wasm_payable_call_via_xvm_works() {
+    new_test_ext().execute_with(|| {
+        let contract_addr = deploy_wasm_contract("payable");
+
+        let prev_balance = Balances::free_balance(&contract_addr);
+        let value = UNIT;
+        assert_ok!(Xvm::call(
+            Context {
+                source_vm_id: VmId::Evm,
+                weight_limit: Weight::from_parts(10_000_000_000, 1024 * 1024),
+            },
+            VmId::Wasm,
+            ALICE,
+            MultiAddress::<AccountId32, ()>::Id(contract_addr.clone()).encode(),
+            // Calling `deposit`
+            hex::decode("0000002a").expect("invalid selector hex"),
+            value
+        ));
+        assert_eq!(
+            Balances::free_balance(contract_addr.clone()),
+            value + prev_balance
+        );
+    });
+}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The test function `wasm_payable_call_via_xvm_works` tests the cross-VM call from the EVM VM to the Wasm VM. The test deploys a Wasm contract, performs a payable call to the contract's `deposit` function, and verifies the balance change. No issues found.



<!-- This is an auto-generated comment by OSS CodeRabbit -->